### PR TITLE
Added Center align title text and image on UIButton method to UIButton extension

### PR DESCRIPTION
**Pull request by imjog**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a changelog entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/UIKit/UIButtonExtensions.swift [hunk 1]
```diff
@@ -161,6 +161,16 @@ public extension UIButton {
...
 		states.forEach { self.setTitle(title, for: $0) }
 	}
+    
+    /// SwifterSwift: Center align title text and image on UIButton
+    ///
+    /// - Parameter spacing: spacing between UIButton title text and UIButton Image.
+    public func centerTextAndImage(spacing: CGFloat) {
+        let insetAmount = spacing / 2
+        imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/UIKit/UIButtonExtensions.swift]: There's some reason for the title insets be the inverse of image insets(image negative on left and title negative on right )? I think it makes the image and goes in different directions increasing the spacing between them.

File: Sources/Extensions/UIKit/UIButtonExtensions.swift [hunk 2]
```diff
@@ -161,6 +161,16 @@ public extension UIButton {
...
+        contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/UIKit/UIButtonExtensions.swift]: I think that in this case, contentEdgeInsets set is not necessary :)

File: Sources/Extensions/UIKit/UIButtonExtensions.swift (hunk 1 above)
Comment (imjog) [reply Sources/Extensions/UIKit/UIButtonExtensions.swift]: yeah! that is purpose of this

File: Sources/Extensions/UIKit/UIButtonExtensions.swift (hunk 2 above)
Comment (imjog) [reply Sources/Extensions/UIKit/UIButtonExtensions.swift]: We can remove `contentEdgeInsets`

**Comment by imjog** [comment]
Hi @LucianoPAlmeida thank you so much for reviewing the PR. For unit test thing - I didn't practices unit test so i am sorry to saying that i coudn't add test for this. can you please help me with this?

**Comment by LucianoPAlmeida** [comment]
Hey @imjog for sure I can help you with this :) Just tell me what you need to know about adding tests for this case and we can work together on adding then :)

File: Sources/Extensions/UIKit/UIButtonExtensions.swift (hunk 1 above)
Comment (LucianoPAlmeida) [reply Sources/Extensions/UIKit/UIButtonExtensions.swift]: Nice :) Now that you say this is the purpose I'm thinking about the naming of this extension, what you think about `textAndImageCenterSpacing(_ spacing: )`? I think it makes the purpose more clear. Let me know what you think :)

Comment (imjog) [reply Sources/Extensions/UIKit/UIButtonExtensions.swift]: When I added this extension, I had two things in mind:
1. By default, if we set an image to button it aligns left but for nice UI we want to Center it just before the title text of the button. So the first purpose of this extension was **Center align Image and Title text on the button**.
2. We want some spacing between title text and image. so second purpose was **Add some spacing between center aligned image and title text**.

So can we name this `centerTextAndImage(with spacing: CGFloat)`?
@LucianoPAlmeida your views?

**Comment by imjog** [comment]
thank you @LucianoPAlmeida :)
 we can add the test for two things:
1. Check if Image and Title Text is center aligned or not.
2. Spacing between Image and title text is equal to given spacing or not.
do you have other things in mind?

**Comment by LucianoPAlmeida** [comment]
Hey @imjog I don't think you need to add checks at this positional level, I think the test here is very simple. You only need to test if the insets(image and text) have the values you expect. Because your implementation only calculate a value to set in the insets and we know the UIKit API takes care of positioning and it's already tested( at least I think it is) you just have to check the insets values to test your calculation. That's what I think :) Let me know your thoughts about it :)

**Comment by imjog** [comment]
Hey @LucianoPAlmeida I agree with you. I will try to to add test for inset. thank you :)

**Comment by SD10** [comment]
@imjog Any update on this one?

**Comment by imjog** [comment]
Hi @LucianoPAlmeida  can you please add test to the extension?

**Comment by LucianoPAlmeida** [comment]
For sure @imjog :)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by LucianoPAlmeida** [comment]
@imjog I couldn't add the project but did implement the test
```swift
    func testCenterTextAndImage() {
        let button: UIButton = UIButton()
        button.centerTextAndImage(spacing: 20)
        XCTAssertEqual(button.imageEdgeInsets, UIEdgeInsets(top: 0, left: -10, bottom: 0, right: 10))
        XCTAssertEqual(button.titleEdgeInsets, UIEdgeInsets(top: 0, left: 10, bottom: 0, right: -10))

    }
```
Can you resolve the conflicts in your fork and update your branch adding it? :)

**Comment by imjog** [comment]
@LucianoPAlmeida  Thank you for adding tests. I will add test to project and resolve conflicts.

**Comment by imjog** [comment]
Hi @LucianoPAlmeida @SD10 I added the test to the extension (Test is passing). Also resolved the conflicts. Please Merge :)

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Comment by imjog** [comment]
@SD10 Thank you so much for inviting me to join the organization. I will make new Extensions and make good contributions to Org.

**Pull request merged**